pymess/backend/push/onesignal.py

```diff
@@ -0,0 +1,56 @@
+from http import HTTPStatus
+from json.decoder import JSONDecodeError
+
+import requests
+from django.utils import timezone
+from django.utils.encoding import force_text
+from onesignal import DeviceNotification, OneSignal
+
+from pymess.backend.push import PushNotificationBackend
+from pymess.config import settings
+from pymess.models import PushNotificationMessage
+from pymess.utils.logged_requests import generate_session
+
+
+class OneSignalPushNotificationBackend(PushNotificationBackend):
+
+    def _is_response_error(self, response):
+        return response.status_code != HTTPStatus.OK
+
+    def _is_response_partial_error(self, response):
+        return not self._is_response_error(response) and response.json().get('errors')
+
+    def _is_errorous(self, response):
+        return self._is_response_error(response) or self._is_response_partial_error(response)
+
+    def _get_error(self, response):
+        return response.json()['errors'] if self._is_errorous(response) else None
+
+    def publish_message(self, message):
+        onesignal_client = OneSignal(settings.PUSH_NOTIFICATION_ONESIGNAL_APP_ID,
+                                     settings.PUSH_NOTIFICATION_ONESIGNAL_API_KEY)
+        onesignal_client.session = generate_session(slug='pymess - OneSignal', related_objects=(message,))
+
+        notification = DeviceNotification(
+            include_external_user_ids=(message.recipient,),
+            contents={'en': message.content},
+            headings={'en': message.heading},
+            data=message.extra_data,
+            url=message.url,
+        )
+
+        try:
+            response = onesignal_client.send(notification)
```

Comment:
to je hrozná knihovna když to vrací response :(

Reply:
Tohle jsem tam dokonce možná udělal já, když jsem se to snažil opravit. Fixnu.